pyomo/contrib/piecewise/tests/test_incremental.py

```diff
@@ -0,0 +1,251 @@
+            self, 'contrib.piecewise.incremental', make_log_x_model_ordered()
+        )
+
+    def test_descend_into_expressions_objective_target(self):
+        ct.check_descend_into_expressions_objective_target(
+            self, 'contrib.piecewise.incremental', make_log_x_model_ordered()
+        )
+
+    @unittest.skipUnless(SolverFactory('gurobi').available(), 'Gurobi is not available')
+    @unittest.skipUnless(SolverFactory('gurobi').license_is_valid(), 'No license')
+    def test_solve_log_model(self):
+        m = make_log_x_model_ordered()
+        TransformationFactory('contrib.piecewise.incremental').apply_to(m)
+        TransformationFactory('gdp.bigm').apply_to(m)
+        SolverFactory('gurobi').solve(m)
+        ct.check_log_x_model_soln(self, m)
+
+    # Failed during development when j1 vertex ordering got broken
+    @unittest.skipUnless(SolverFactory('gurobi').available(), 'Gurobi is not available')
+    @unittest.skipUnless(SolverFactory('gurobi').license_is_valid(), 'No license')
+    def test_solve_product_model(self):
+        m = ConcreteModel()
+        m.x1 = Var(bounds=(0.5, 5))
+        m.x2 = Var(bounds=(0.9, 0.95))
+        pts = list(itertools.product([0.5, 2.75, 5], [0.9, 0.925, 0.95]))
+        m.pwlf = PiecewiseLinearFunction(
+            points=pts,
+            function=lambda x, y: x * y,
+            triangulation=Triangulation.OrderedJ1,
+        )
+        m.obj = Objective(sense=minimize, expr=m.pwlf(m.x1, m.x2))
+        TransformationFactory("contrib.piecewise.incremental").apply_to(m)
+        SolverFactory('gurobi').solve(m)
+        self.assertAlmostEqual(0.45, value(m.obj))
+
+
+# Make a version of the log_x model with the simplices properly ordered for the
+# incremental transform
+def make_log_x_model_ordered():
+    m = ConcreteModel()
+    m.x = Var(bounds=(1, 10))
+    m.pw_log = PiecewiseLinearFunction(points=[1, 3, 6, 10], function=log)
+
+    # Here are the linear functions, for safe keeping.
+    def f1(x):
+        return (log(3) / 2) * x - log(3) / 2
+
+    m.f1 = f1
+
+    def f2(x):
+        return (log(2) / 3) * x + log(3 / 2)
+
+    m.f2 = f2
+
+    def f3(x):
+        return (log(5 / 3) / 4) * x + log(6 / ((5 / 3) ** (3 / 2)))
+
+    m.f3 = f3
+
+    m.log_expr = m.pw_log(m.x)
+    m.obj = Objective(expr=m.log_expr)
+
+    m.x1 = Var(bounds=(0, 3))
+    m.x2 = Var(bounds=(1, 7))
+
+    ## approximates paraboloid x1**2 + x2**2
+    def g1(x1, x2):
+        return 3 * x1 + 5 * x2 - 4
+
+    m.g1 = g1
+
+    def g2(x1, x2):
+        return 3 * x1 + 11 * x2 - 28
+
+    m.g2 = g2
+    # order for incremental transformation
+    simplices = [
+        [(0, 1), (3, 1), (3, 4)],
+        [(3, 4), (0, 1), (0, 4)],
+        [(0, 4), (0, 7), (3, 4)],
+        [(3, 4), (3, 7), (0, 7)],
+    ]
+    m.pw_paraboloid = PiecewiseLinearFunction(
+        simplices=simplices,
+        linear_functions=[g1, g1, g2, g2],
+        override_triangulation=Triangulation.AssumeValid,
```

Comment:
So why not let people specify `Triangulation.AssumeValid` *as* the triangulation, if they want to use their own simplices. We could even require that,  which then means that we could give a proper warning when they have both simplices and Delaunay explicitly specified. I think that would be cleaner than adding an additional argument.